### PR TITLE
Make gflags an IMPORTED GLOBAL library

### LINKED ISSUE #1197: Importing libraries via thirdparty makes libjxl impossible to build on Windows
**Describe the bug**
I cannot build libjxl from scratch on Windows due to [CMake issue 22919](https://gitlab.kitware.com/cmake/cmake/-/issues/22919). This affects specifically the gflags dependency; since libjxl imports libraries in a sibling directory to `tools`,  `gflags`'s location cannot be seen at link time:

(excerpt of `build\tools\cjxl_ng.vcxproj)
```xml
<Link>
      <AdditionalDependencies>..\lib\MinSizeRel\jxl.lib;gflags.lib;box\MinSizeRel\box.lib;..\lib\MinSizeRel\jxl-static.lib;..\lib\MinSizeRel\jxl_extras-static.lib;..\lib\MinSizeRel\jxl_threads-static.lib;MinSizeRel\jxl_tool.lib;..\lib\MinSizeRel\jxl_extras_dec-static.lib;E:\krita-win\i_deps_msvc\lib\gif.lib;E:\krita-win\i_deps_msvc\lib\jpeg.lib;E:\krita-win\i_deps_msvc\lib\libpng16.lib;E:\krita-win\i_deps_msvc\lib\zlib.lib;..\lib\MinSizeRel\jxl-static.lib;E:\krita-win\i_deps_msvc\lib\brotlidec.lib;E:\krita-win\i_deps_msvc\lib\brotlicommon.lib;E:\krita-win\i_deps_msvc\lib\hwy.lib;E:\krita-win\i_deps_msvc\lib\brotlienc.lib;E:\krita-win\i_deps_msvc\lib\lcms2.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;comdlg32.lib;advapi32.lib</AdditionalDependencies>
```

(excerpt of build log):
```
[build] LINK : fatal error LNK1104: no se puede abrir el archivo 'gflags.lib' [E:\libjxl\build\tools\cjxl_ng.vcxproj]
```

**To Reproduce**

Build and install all dependencies separately: brotli gflags giflib highway jpeg lcms2 openexr png

Then configure and build libjxl with e.g.:

```json
{
    "cmake.configureSettings": {
        "CMAKE_PREFIX_PATH": "E:/krita-win/i_deps_msvc",
        "JPEGXL_VERSION": "0.6.1",
        "BUILD_TESTING": false,
        "JPEGXL_ENABLE_BENCHMARK": false,
        "JPEGXL_ENABLE_COVERAGE": false,
        "JPEGXL_ENABLE_EXAMPLES": false,
        "JPEGXL_ENABLE_FUZZERS": false,
        "JPEGXL_ENABLE_JNI": false,
        "JPEGXL_ENABLE_SJPEG": false,
        "JPEGXL_ENABLE_SKCMS": false,
        "JPEGXL_ENABLE_VIEWERS": false
    },
}
```

**Expected behavior**
The library is correctly built and installed.

**Screenshots**
N/A

**Environment**
 - OS: [e.g. Windows]: Windows 10 20H2
 - Compiler version: [e.g. clang 11.0.1] MSVC 19.29.30140
 - CPU type: [e.g. x86_64] x86_64
 - cjxl/djxl version string: [e.g. cjxl [v0.3.7 | SIMD supported: SSE4,Scalar]] commit ea6a190cd777b61662bd0470db24163a1a63bdab

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by amyspark**
See:

- https://github.com/microsoft/vcpkg/pull/20011

- https://stackoverflow.com/questions/45401212/how-to-make-imported-target-global-afterwards/48390363#48390363

Fixes #1197

### CODE REVIEW

**Comment by mo271** [comment]
Thanks for the contribution! Could you add your name to the [AUTHORS] file? (https://github.com/libjxl/libjxl/blob/42b66f16faaa465a6fc61eb9e89f4f3327cfa810/AUTHORS )

File: third_party/CMakeLists.txt [hunk 1]
```diff
@@ -179,6 +179,8 @@ if(JPEGXL_ENABLE_TOOLS)
                      ${PROJECT_BINARY_DIR}/LICENSE.gflags COPYONLY)
     endif()  # JPEGXL_DEP_LICENSE_DIR
   endif()
+  # https://gitlab.kitware.com/cmake/cmake/-/issues/22919
+  set_target_properties(gflags PROPERTIES IMPORTED_GLOBAL TRUE)
```
Comment (mo271) [inline third_party/CMakeLists.txt:183]: Getting cmake error "set_target_properties can not be used on an ALIAS target.".
Perhaps that can easily be fixed by move those two lines inside the ifs above?!

**Review by mo271 — DISMISSED** [review]

**Comment by mo271** [comment]
Thanks for modifying AUTHORS! This check is green now.

**Comment by sboukortt** [comment]
Thanks!

**Review by sboukortt — APPROVED** [review]

**Comment by jonsneyers** [comment]
This broke something for the [Release linux x86_64 static](https://github.com/libjxl/libjxl/runs/6075456254?check_suite_focus=true#logs) build:

```
[288/310] Linking CXX executable tools/cjxl_ng
FAILED: tools/cjxl_ng 
: && /usr/bin/clang++ -ffunction-sections -fdata-sections -fno-rtti -funwind-tables -Xclang -mrelax-all -Xclang -mconstructor-aliases -fno-omit-frame-pointer -O3 -DNDEBUG -O2 -Wl,--gc-sections -Wl,--no-export-dynamic -static -static-libgcc -static-libstdc++    -fPIE -pie tools/CMakeFiles/cjxl_ng.dir/cjxl_ng_main.cc.o -o tools/cjxl_ng  lib/libjxl.a  lib/libjxl_extras_dec-static.a  lib/libjxl_threads.a  third_party/highway/libhwy.a  /usr/lib/x86_64-linux-gnu/libgflags.so.2.2.2  tools/box/libbox.a  lib/libjxl.a  lib/libjxl_extras-static.a  lib/libjxl_threads.a  tools/libjxl_tool.a  lib/libjxl_extras_dec-static.a  /usr/lib/x86_64-linux-gnu/libgif.a  /usr/lib/x86_64-linux-gnu/libjpeg.a  /usr/lib/x86_64-linux-gnu/libpng.a  /usr/lib/x86_64-linux-gnu/libz.a  third_party/sjpeg/libsjpeg.a  lib/libjxl.a  third_party/highway/libhwy.a  third_party/brotli/libbrotlidec-static.a  -lpthread  -Wl,--whole-archive  -Wl,--no-whole-archive  third_party/brotli/libbrotlienc-static.a  third_party/brotli/libbrotlicommon-static.a  -lm && :
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
/usr/bin/ld: attempted static link of dynamic object `/usr/lib/x86_64-linux-gnu/libgflags.so.2.2.2'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by mo271** [comment]
> This broke something for the [Release linux x86_64 static](https://github.com/libjxl/libjxl/runs/6075456254?check_suite_focus=true#logs) build:
> 
> ```
> [288/310] Linking CXX executable tools/cjxl_ng
> FAILED: tools/cjxl_ng 
> : && /usr/bin/clang++ -ffunction-sections -fdata-sections -fno-rtti -funwind-tables -Xclang -mrelax-all -Xclang -mconstructor-aliases -fno-omit-frame-pointer -O3 -DNDEBUG -O2 -Wl,--gc-sections -Wl,--no-export-dynamic -static -static-libgcc -static-libstdc++    -fPIE -pie tools/CMakeFiles/cjxl_ng.dir/cjxl_ng_main.cc.o -o tools/cjxl_ng  lib/libjxl.a  lib/libjxl_extras_dec-static.a  lib/libjxl_threads.a  third_party/highway/libhwy.a  /usr/lib/x86_64-linux-gnu/libgflags.so.2.2.2  tools/box/libbox.a  lib/libjxl.a  lib/libjxl_extras-static.a  lib/libjxl_threads.a  tools/libjxl_tool.a  lib/libjxl_extras_dec-static.a  /usr/lib/x86_64-linux-gnu/libgif.a  /usr/lib/x86_64-linux-gnu/libjpeg.a  /usr/lib/x86_64-linux-gnu/libpng.a  /usr/lib/x86_64-linux-gnu/libz.a  third_party/sjpeg/libsjpeg.a  lib/libjxl.a  third_party/highway/libhwy.a  third_party/brotli/libbrotlidec-static.a  -lpthread  -Wl,--whole-archive  -Wl,--no-whole-archive  third_party/brotli/libbrotlienc-static.a  third_party/brotli/libbrotlicommon-static.a  -lm && :
> clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
> /usr/bin/ld: attempted static link of dynamic object `/usr/lib/x86_64-linux-gnu/libgflags.so.2.2.2'
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```

Seems to be fixed here https://github.com/libjxl/libjxl/pull/1364

**Pull request merged**